src/Microsoft.ApplicationInsights/Extensibility/Implementation/Endpoints/ConnectionStringDuplicateKeyException.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Endpoints
+{
+    using System;
+#if !NETSTANDARD1_3
+    using System.Runtime.Serialization;
+#endif
+
+    /// <summary>
+    /// This exception is used to notify the user that their Connection String has duplicate keys.
+    /// </summary>
+#if !NETSTANDARD1_3
+    [Serializable]
+#endif
+    public class ConnectionStringDuplicateKeyException : Exception
```

Comment:
Curious, what's the benefit of having concrete exception types deriving from `Exception`?
What's the pros / cons of using one exception type for all the connection string validation failures?

Reply:
My intention here was to be explicit with what exception is being thrown.
Also In the unit tests, this lets me catch the type instead of evaluating the message.

Reply:
What's the benefit of telling the concrete error from unit test?
For example, we could potentially do:
1. feed the unit test with a valid connection string, expect a valid result.
2. feed the unit test with an invalid connection string, expect an `ArgumentException`.

Reply:
It's no benefit to the unit test.
The test is validating that a meaningful exception was emitted. 

I can rework this to use `ArgumentException`.